shared_model/bindings/model_proto.hpp

```diff
@@ -38,9 +38,13 @@ namespace shared_model {
        * @param keypair - keypair to sign
        * @return blob of signed transaction
        */
-      crypto::Blob signAndAddSignature(Unsigned &us,
-                                       const crypto::Keypair &keypair) {
-        return us.signAndAddSignature(keypair).blob();
+      Unsigned signAndAddSignature(Unsigned &us,
+                                   const crypto::Keypair &keypair) {
+        return us.signAndAddSignature(keypair);
```

Comment:
This doesn't work as expected: after this call client gets object of `Unsigned` type (not `ModelProto<Unsigned>`). Next calls to `signAndAddSignature` and `finish` will call methods from `UnsignedWrapper` directly. That leads to double `blob()` calls since `ModelProto<Unsigned>` calls `blob()` by itself and `UnsignedWrapper` doesn't.
I suggest to change `signAndAddSignature` in `ModelProto` to return instance of `ModelProto<Unsigned>` instead of `UnsignedWrapper`.

Reply:
Thanks!